feat(Configuration): add support for ember serve run configuration

makepanic opened the pull request:
This PR adds a `Ember serve` run configuration and should fix #155.

Currently it's pretty barebone and only generates an `ember s` command.
All defaults and fields are hardcoded.

I kinda dislike the swing form code and we can probably refactor it to make it more compact.

Turbo87 commented on .gitignore:
```diff
@@ -30,3 +30,5 @@ gradle-app.setting
 # Cache of project
 .gradletasknamecache
+
+.idea
```
I'm not a fan of adding editor-specific things to each project `.gitignore` file. I'd suggest to use a global `.gitignore` file for these things instead. (see https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

makepanic replied:
Thanks, I didn't know that a global ignore was possible. I removed the gitignore change.

Turbo87 commented:
I got a few thoughts on the config page:

- "Environment" should probably be a radio button or dropdown list of `development` and `production`
- Everything aside from "Environment" and "Port" seems like advanced settings to me that the average user does not use and probably should not use without knowing what exactly they're doing. Can we add a collapsible region (like the "Before Launch" thing at the bottom) called "Advanced Settings" where those settings live?
- Why do we have the single "Configuration" tab at the bottom? Since it's the only tab we probably don't need it, right?
- "Output path" has `dist/` by default, and I assume that means that we use `--output-path=dist/`. We should probably remove the default value from that field and if the field is empty not add the `--output-path` option at all. We could still show it as a placeholder maybe. (Same for "Port", "Watcher", etc.)

In general this seems to work quite well though. Very happy to see the progress on this :)

makepanic commented:
> "Environment" should probably be a radio button or dropdown list of development and production

I've changed it to a `JComboBox` which [contains](https://github.com/Turbo87/intellij-emberjs/pull/165/commits/6631169f83783493b278a8e0ec0fb2ad6fdd4003#diff-f942c7b140bf2172831e6ad10e45f490R112) options for `development` and `production`.

> Everything aside from "Environment" and "Port" seems like advanced settings to me that the average user does not use and probably should not use without knowing what exactly they're doing. Can we add a collapsible region (like the "Before Launch" thing at the bottom) called "Advanced Settings" where those settings live?

I've added that one

> Why do we have the single "Configuration" tab at the bottom? Since it's the only tab we probably don't need it, right?

You're right. I've added it because most other configurations have at least a `Configuration` and `Logs` tab. I guess we could add it if needed.

> "Output path" has dist/ by default, and I assume that means that we use --output-path=dist/. We should probably remove the default value from that field and if the field is empty not add the --output-path option at all. We could still show it as a placeholder maybe. (Same for "Port", "Watcher", etc.)

Using `EditorTextField`, that's now implemented.

Turbo87 commented:
awesome! could you add a section about that feature to https://github.com/Turbo87/intellij-emberjs/blob/master/doc/features.md with a screenshot of the dialog?

Turbo87 commented:
also I think we should call it either `ember serve` or `Ember Serve`, but not `Ember serve`

Turbo87 commented:
<img width="1183" alt="bildschirmfoto 2017-10-22 um 14 55 33" src="https://user-images.githubusercontent.com/141300/31862115-2aa18638-b739-11e7-931c-cd5fea1faa4e.png">

makepanic commented:
I've renamed the configuration type name and added a small section to the features document.
The screenshot looks pretty plain because it wasn't done on macOS and it has no window decorations enabled.

Turbo87 commented:
thanks!!!!!!

Turbo87 commented:
releasing as `v2017.3.2` :)
